### PR TITLE
Fix filters

### PR DESCRIPTION
**Pull request by bienzaaron**
Issue #99, #98, and #97 handled. 

Rather than having an entire class devoted to filtering, I just put a filter method in MachineAdapter, since filtering only on if a machine is available or not is much simpler. 

I separated the network request from updating the recycler view to get rid of the unnecessary network request.

Also, the notify button now appears when there are no machines at all, not just when there are no machines visible.

### CODE REVIEW

**Comment by hughesjeff** [comment]
So I'm just gonna make a note here saying why #97 exists. 

So, in the AppSectionsPagerAdapter, when notifyDataSetChanged was called, it's data set is the two fragments. Down the chain, it recreates the Fragments! This of course goes down and calls refreshList(), which is where the network call comes from. 

You can see this by adding notifyDataSetChanged in the beginning of notifyFilterChanged and adding a debug breakpoint somewhere in onCreateView in MachineFragment. 

This is because notifyDataSetChanged in AppSectionsPagerAdapter's adapter is actually the Fragments, so it's gonna go and update those instead of RecyclerView adapters in the Fragments themselves.

Specifically, when in getItemPosition, you return POSITION_NONE, that's a signal to always recreate the Fragments! API Reference: https://developer.android.com/reference/android/support/v4/view/PagerAdapter.html#getItemPosition(java.lang.Object)

The first reply here also has some more info about it: https://stackoverflow.com/questions/16671695/how-to-recreate-all-fragments-in-viewpager

Fun stuff!

TL;DR: notifyDataSetChanged in AppSectionsPagerAdapter works on the Fragments, not the RecyclerViews.

**Comment by hughesjeff** [comment]
Also, congrats on PR 100!

**Review by hughesjeff — APPROVED** [review]
